gr-fec/lib/generic_decoder.cc

```diff
@@ -8,21 +8,23 @@
  *
  */
 
+#include <memory>
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
 
 #include <gnuradio/fec/generic_decoder.h>
 
+#include <memory>
+
 namespace gr {
 namespace fec {
 
 generic_decoder::generic_decoder(std::string name)
 {
     d_name = name;
     my_id = base_unique_id++;
-
-    GR_LOG_ASSIGN_CONFIGURED_LOGPTR(d_logger, alias());
+    d_logger = std::make_shared<gr::logger>(alias());
```

Comment:
The macro is silly, but a `gr::logger::make()` factory might be nice.

Reply:
ah to get a shared ptr right away? Didn't do that, because I wanted to move away from these, in the long term: a logger is pretty much a privately held object. Here, it makes little sense to restructure the whole thing just to make this happen right away, but plain objects or unique ptrs make more sense – the logging itself is thread-safe.

Reply:
hm this definitely sounds like something I should document as design rationale; else, I might as well add a factory, you're right, that would've been nicer.